### PR TITLE
[Fix] Preserve sampler customized_info across PD disaggregation

### PR DESCRIPTION
**Pull request by aurickq**
## Motivation

In disaggregated serving, prefill samples the first output token, but its sampler-provided `customized_info` does not reach decode. This leaves per-token metadata shorter than the generated token sequence.

## Modifications

- Collect the prefill token's `customized_info` and transfer it through the existing metadata buffers, then append it to the decode request. Skip fake transfers and replayed boundaries, and clear metadata when prefill is retried.
- Use MessagePack with the existing NumPy/tensor codecs; tensors are serialized on CPU so the sender's GPU ordinal is not reused on the decode host. Preserve zero-dimensional NumPy array shapes.
- Default to a 4 KiB payload buffer per request. `SGLANG_DISAGG_CUSTOMIZED_INFO_MAX_BYTES` overrides the capacity; oversized or unencodable payloads abort only the affected request through the existing transfer-failure cleanup. Prefill and decode must use matching builds and the same capacity setting; capacity equality is not negotiated or checked. NIXL also rejects mismatched metadata-buffer counts before transfer.
- Reuse the existing customized-info sampler probe in the unified-memory PD logprob-parity tests, checking scalar and array metadata against every output token, including the first. Add focused buffer roundtrip/capacity checks.

## Accuracy Tests

- Focused prefill-abort, disaggregation wire, NIXL backend, and MessagePack IPC unit suites: 125 passed, plus 40 subtests. The request-local failure regression covers overflow and unsupported values, one-time cleanup, a subsequent valid send, and propagation of unexpected runtime errors; both payload-error cases fail against the original handler.
- Changed-file pre-commit checks: passed.
- Upstream CI passed the unified-memory PD parity, optimistic-prefill, and customized-info streaming suites on the previous revision. GPU validation of the updated revision is pending.

## Speed Tests and Profiling

No performance benchmark. Cross-node RDMA has not been validated locally.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #35554804109](https://github.com/sgl-project/sglang/actions/runs/35554804109)
Latest PR Test (Extra): :white_check_mark: [Run #35554803983](https://github.com/sgl-project/sglang/actions/runs/35554803983)
Latest PR Test (AMD ROCm 10): :x: [Run #35554803909](https://github.com/sgl-project/sglang/actions/runs/35554803909)

### CODE REVIEW

**Comment by ispobock** [comment]
/rerun-test test/registered/disaggregation/test_disaggregation_unified_memory.py test/registered/unit/managers/test_customized_info_streaming.py test/registered/unit/disaggregation/test_disaggregation_wire.py test/registered/unit/disaggregation/test_nixl_backend_basic.py test/registered/unit/managers/test_msgpack_ipc_roundtrip.py test/registered/disaggregation/test_disaggregation_optimistic_prefill.py test/registered/disaggregation/test_disaggregation_nixl.py

File: python/sglang/srt/disaggregation/utils.py [hunk 1]
```diff
@@ -451,7 +475,42 @@ def get_buf(self, idx: int):
...
+            else msgspec.msgpack.encode(
+                req.customized_info,
+                # Do not carry the sender's GPU ordinal to the decode host.
+                enc_hook=lambda obj: enc_hook(
+                    obj.detach().cpu() if isinstance(obj, torch.Tensor) else obj
+                ),
+            )
+        )
+        if len(payload) > self.output_customized_info.shape[1]:
+            raise ValueError(
```
Comment (ispobock) [inline python/sglang/srt/disaggregation/utils.py]: Could we abort only the affected request here? This uncaught `ValueError` reaches the scheduler’s fatal handler and shuts down the prefill service.

Comment (aurickq) [reply python/sglang/srt/disaggregation/utils.py]: Fixed in 9f3e5f4de. Oversized payloads and expected encoding failures now abort only the affected request. The sender is marked failed, and the request stays in the inflight queue for the existing rank-consistent cleanup; unrelated exceptions still propagate.

**Comment by ispobock** [comment]
some cpu tests failed https://github.com/sgl-project/sglang/actions/runs/35505113329/job/106063585551

**Comment by aurickq** [comment]
Fixed in 6e4d2756c. I reproduced the four failures across the three failed CPU shards: two existing mocks were missing the new customized-info methods, and the abort regression needed the current get_parallel().override(tp_rank=0) fixture instead of scheduler.ps after the merge from main.

This is a test-only fix (6 insertions, 4 deletions), with the existing assertions unchanged. The three affected suites plus the wire-format, NIXL-backend, and MessagePack IPC suites pass locally with GPUs hidden: 169 tests and 44 subtests. Changed-file pre-commit checks also pass; full CPU CI still needs to rerun.

File: python/sglang/srt/disaggregation/nixl/conn.py [hunk 2]
```diff
@@ -2259,6 +2259,16 @@ def send_aux(
 
         prefill_aux_ptrs = self.kv_args.aux_data_ptrs
         prefill_aux_item_lens = self.kv_args.aux_item_lens
+        if not (
+            len(prefill_aux_ptrs) == len(prefill_aux_item_lens) == len(dst_aux_ptrs)
```
Comment (ispobock) [inline python/sglang/srt/disaggregation/nixl/conn.py:2263]: this checks buffer count, not per-slot size. a P/D `SGLANG_DISAGG_CUSTOMIZED_INFO_MAX_BYTES` mismatch passes here, then send_aux strides by prefill's item_len into a smaller decode slot and clobbers neighbor requests. exchange aux_item_lens at handshake and refuse on mismatch.
